src/runtime/aot_executor/aot_executor.cc

```diff
@@ -62,9 +64,38 @@ AotExecutor::AotExecutor(tvm::runtime::Module module, const std::vector<Device>&
                                       output->dtype(), devices_[0]));
   }
 
-  for (auto pool : metadata_->pools()) {
-    args_.emplace_back(NDArray::Empty(ShapeTuple(pool->shape().begin(), pool->shape().end()),
-                                      pool->dtype(), devices_[0]));
+  auto get_array_byte_size = [](DataType data_type, auto shape) {
+    int64_t nbytes = (data_type.bits() * data_type.lanes() + 7) / 8;
```

Comment:
could you reuse one of the existing functions we have for computing array extent size?

Reply:
reused